### ISSUE TITLE
Amarok not working with phonon + gstreamer + pulseaudio + kde5

### ISSUE THREAD

**Issue by jerith666** [NixOS/nixpkgs#16588]
## Issue description

I can't seem to make Amarok play any music.  The initial symptom was that `PHONON_BACKEND_DEBUG=3 PHONON_SUBSYSTEM_DEBUG=1 amarok --debug` printed:

```
"Warning: You do not seem to have the base GStreamer plugins installed.
All audio and video support has been disabled"
Phonon::GStreamer::Backend: Failed to initialize GStreamer
```

... the play controls looked to be drawn incompletely (only a play button, no skip fwd/back, etc.), and the phonon settings dialog was completely empty.

After some digging, I hit upon the distinction between `GST_PLUGIN_SYSTEM_PATH` and `GST_PLUGIN_SYSTEM_PATH_1_0`.  Only the latter was set in my environment, because I'm using kde5 and its nix expression (at `nixos/modules/services/x11/desktop-managers/kde5.nix`) only sets that.  However, amarok looked to be wanting the former since it depends on gstreamer-0.10:

```
$ nix-store -q --graph $(dirname $(dirname $(readlink -e $(type -p amarok)))) | grep 'gst.*->'
"/nix/store/vf7n11vh8sksa70gp3l2bsvv3h3j0f3p-gst-plugins-base-0.10.36" -> "/nix/store/i8r91xw4b5bg0349kdc7hr4pc63cnxqf-qt-4.8.7" [color = "red"];
"/nix/store/vj9jznlp4y6s2c3c8vnc7him20i6iiim-gstreamer-0.10.36" -> "/nix/store/i8r91xw4b5bg0349kdc7hr4pc63cnxqf-qt-4.8.7" [color = "green"];
"/nix/store/vj9jznlp4y6s2c3c8vnc7him20i6iiim-gstreamer-0.10.36" -> "/nix/store/vf7n11vh8sksa70gp3l2bsvv3h3j0f3p-gst-plugins-base-0.10.36" [color = "burlywood"];
```

So I added this to `configuration.nix` (cribbed from `kde4.nix`):

```
environment.profileRelativeEnvVars = {
  GST_PLUGIN_SYSTEM_PATH = [ "/lib/gstreamer-0.10" ];
};
```

and added `gst_all.gstPluginsBase` to `environment.systemPackages`.  This instead yielded a crash in `Phonon::Gstreamer::MediaNode::linkMediaNodeList`.  Adding `gst_all.gstPluginsGood gst_all.gstPluginsBad gst_all.gstPluginsUgly` didn't help.  Flailing some more, I added `gstreamer`; that turned the crash into a hang: the amarok window appears but does not draw completely.

At this point I _think_ that the problem is that `pkgs/development/libraries/phonon/backends/gstreamer.nix` depends on gstreamer 1.x, while amarok depends on gstreamer 0.x.  I tried tweaking the former to use `gst_all` instead of `gst_all_1`, but that produced a build failure:

```
-- Checking for module 'gstreamer-1.0'
--   Package 'gstreamer-1.0' not found
```

It doesn't look easily possible to build amarok against `gst_all_1`, since its dependency comes indirectly via `qt`.

I'm hoping for some help from @ttuegel, whose name is all over the relevant git logs.  :)
## Technical details
- System: `16.09.git.53c0a7d (Flounder)` (based on `nixos-unstable` @ 453086a)
- Nix version: `nix-env (Nix) 1.11.2`
- Nixpkgs version: `16.09.git.5431facM`

**Comment by joelmo** [comment]
I am not a amarok user but I see same issue here, actually not anything but the interface appears to be working.

**Comment by jerith666** [comment]
There is another fix needed to make the collection work with MySQL / MariaDB: https://github.com/NixOS/nixpkgs/pull/16578

**Comment by jerith666** [comment]
Just an update on the current state: seems to be about the same as before.  On current `nixos-unstable` (@2839b101f9), `PHONON_BACKEND_DEBUG=3 PHONON_SUBSYSTEM_DEBUG=1 amarok --debug` still prints:

```
amarok:     BEGIN: void EngineController::initializePhonon() 
PHONON-GST       [WARNING] "Warning: You do not seem to have the base GStreamer plugins installed.
          All audio and video support has been disabled" 
PHONON-GST       [WARNING] "Warning: You do not seem to have the base GStreamer plugins installed.
          All audio and video support has been disabled" 
PHONON-GST       Using GStreamer 0.10.36 
Phonon::GStreamer::Backend: Failed to initialize GStreamer
PHONON-GST       [WARNING] Backend class 2 is not going to be created because GStreamer init failed. 
PHONON-GST       [WARNING] Backend class 3 is not going to be created because GStreamer init failed. 
PHONON-GST       [WARNING] Backend class 0 is not going to be created because GStreamer init failed. 
WARNING: Phonon::createPath: Cannot connect  Phonon::MediaObject ( no objectName ) to  Phonon::AudioDataOutput ( no objectName ). 
PHONON-GST       [WARNING] Backend class 0 is not going to be created because GStreamer init failed. 
WARNING: Phonon::createPath: Cannot connect  Phonon::MediaObject ( no objectName ) to  Phonon::AudioOutput ( no objectName ). 
PHONON-GST       [WARNING] Backend class 0 is not going to be created because GStreamer init failed. 
PHONON-GST       [WARNING] Backend class 0 is not going to be created because GStreamer init failed. 
PHONON-GST       [WARNING] Backend class 1 is not going to be created because GStreamer init failed. 
PHONON-GST       [WARNING] Backend class 1 is not going to be created because GStreamer init failed. 
PHONON-GST       [WARNING] Backend class 0 is not going to be created because GStreamer init failed. 
amarok:       [EngineController] Tick Interval (actual):  100 
PHONON-GST       [WARNING] Backend class 0 is not going to be created because GStreamer init failed. 
amarok:     END__: void EngineController::initializePhonon() [Took: 0.076s] 
```

... but the UI is responsive.  With the `environment.profileRelativeEnvVars` and `environment.systemPackages` changes from the description reapplied, amarok still hangs on startup.

**Comment by jerith666** [comment]
It looks like amarok has been removed in 6c317b62a3b8eb5007b7655640309441b449dd7b / #32927.  There is still amarok-kf5 (#26436), but I haven't tested that.

However, in the course of researching it, I discovered https://www.clementine-player.org/, which is in nixpkgs and after a bit of testing seems to do the main things I used amarok for.  So, if others are running into this problem, I recommend trying clementine as a replacement.
